numpy/core/src/multiarray/methods.c

```diff
@@ -1822,7 +1823,7 @@ array_setstate(PyArrayObject *self, PyObject *args)
             }
             if (swap) {
                 /* byte-swap on pickle-read */
-                npy_intp numels = num / PyArray_DESCR(self)->elsize;
+                npy_intp numels = PyArray_SIZE(self);
```

Comment:
Seems sensible, but why was it done differently before?

Reply:
To avoid recalculating `num`, most likely